### ISSUE TITLE
Object detection using masks RCNN : OOM errors

### ISSUE THREAD

**Issue by WbHappy** [cagbal/ros_people_object_detection_tensorflow#33]
Camera : ZED Stereo camera from StereoLabs (resolution : 1280*720 | 30 images per seconds)
Graphical card : Nvidia GTX 760 2GB Video RAM
Model name : mask_rcnn_inception_v2_coco_2018_01_28
Issue : _ResourceExhaustedError: OOM when allocating tensor with shape[100,1024,15,15]_

Hello,

First of all, let met thank you for this Tensorflow ROS wrapper ! I managed to install it and use it quite easily using your instructions.

I am trying to move forward and detect object not only using bounding boxes, which works well, but directly using masks (using the model :  mask_rcnn_inception_v2_coco_2018_01_28). I am getting some images (like one every 15 seconds) with masks but after some time I get the above error.

I know that I don't have the newest graphical card and that its video RAM is really low but I am just puzzled that using the _ssd_mobilenet_v1_coco_2017_11_17_ model it works fine (only bounding boxes).

Googling my issue allowed me to see that reducing the batch_size can help solving those OOM errors. However this has a meaning only in the training part.. 

My question here is do you have some tips on how to improve the time processing (to get more image tagged per second) and remove those OOM errors ?

Thank you for your answers.

**Comment by cagbal** [comment]
Hi! 
I am happy that the repo works for you. I did not get this issue before, I will try to reproduce it.

**Comment by Wagdi-Benyaala** [comment]
Thanks for the fast answer !

Well actually without GPU I am getting fast result for bounding boxes and slow results (7-15sec) for masks, with no crashes.

Using GPU (NVIDIA GTX 760 with Cuda 9.2 and tensorflow1.8-gpu) I get really good results for bounding boxes but also slow results (about the same ~15sec) for masks... but getting a crash OOM at the end.
